fix(integrations): ASGI integration not capture transactions in Websocket

guodong000 opened the pull request:
fix(integrations): ASGI integration not capture transactions in Websocket

In [ASGI Specs](https://github.com/django/asgiref/blob/main/specs/www.rst#websocket-connection-scope), `method` is not in Websocket Connection Scope.

sentrivana commented:
Hey @guodong000, thank you for this fix -- the current behavior definitely looks incorrect.

I'd strongly suggest we add a test case for this so that we don't regress (if we had this in the first place, this never would have broken). Do you have some time to add one to `tests/integrations/test_asgi.py`? There's already a `test_websocket` in the file that tests whether we capture errors -- this can likely be used as a base to create a new test case for testing transactions with some small effort.

Let me know -- if you don't have the time we can take care of it.

guodong000 commented:
I found that the `test_websocket` has some issues and fixed it.

I checked the code of `async-asgi-testclient`, The `TestClient` actually uses the Lifespan protocol in context manager methods.

Then I added some related assertions.

sentrivana commented:
Can't comment on the specific line because it's not part of the diff but you probably need to change the `sentry_init` in the test to
```python
sentry_init(send_default_pii=True, traces_sample_rate=1.0)
```
And then you should (if everything is working) actually get a separate transaction event [here](https://github.com/getsentry/sentry-python/pull/4293/files#diff-a4ca49dac9c5fcce27a4fb5fb53e5f8e764b807773ba7c5476d70d74a9789353R365) in addition to the message and the error. As is, the test is checking whether the message is associated with a transaction, but not if a transaction event is actually being emitted.

guodong000 commented:
@sentrivana I have made the changes according to your suggestions. Please take a look.

sentrivana reviewed (approved):
Thank you @guodong000!
